### ISSUE TITLE
Failure to parse CREATE VIEW with T-SQL dialect

### ISSUE THREAD

**Issue by borchero** [sqlfluff/sqlfluff#3667]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

`CREATE VIEW` statements cannot be parsed (when the dialect is set to `tsql`).

### Expected Behaviour

The query should be parsed without any issues (see [documentation](https://docs.microsoft.com/en-us/sql/t-sql/statements/create-view-transact-sql?view=sql-server-ver16)).

### Observed Behaviour

Output of `sqlfluff lint <file>`:

```
== [test.sql] FAIL                                                                                                                                    
L:   1 | P:   1 |  PRS | Line 1, Position 1: Found unparsable section: 'CREATE
                       | VIEW my_view (\n my_pk,\n my_...'
WARNING: Parsing errors found and dialect is set to 'tsql'. Have you configured your dialect correctly?
All Finished 📜 🎉!
```

### How to reproduce

```sql
CREATE VIEW my_view (
    my_pk,
    my_col
)
AS 
SELECT
    my_pk,
    my_col 
FROM my_table;
```

### Dialect

T-SQL

### Version

- `sqlfluff, version 1.2.1` (installed via homebrew)

### Configuration

```
[sqlfluff]
dialect = tsql
```

### Are you willing to work on and submit a PR to address the issue?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by tunetheweb** [comment]
At the moment SQLFluff doesn't recognise the optional column name syntax.

Should be easy enough to add by changing this:

```py
    type = "create_view_statement"
    match_grammar = Sequence(
        "CREATE",
        Sequence("OR", "ALTER", optional=True),
        "VIEW",
        Ref("ObjectReferenceSegment"),
        Sequence(
```

to this:

```py
    type = "create_view_statement"
    match_grammar = Sequence(
        "CREATE",
        Sequence("OR", "ALTER", optional=True),
        "VIEW",
        Ref("ObjectReferenceSegment"),
        Bracketed(
            Delimited(
                Ref("IndexColumnDefinitionSegment"),
            ),
            optional=True,
        ),
        Sequence(
```
